client/lib/analytics/index.js

```diff
@@ -559,6 +561,19 @@ const analytics = {
 		}
 	},
 
+	collectCoupon: function() {
+		// When we have `?coupon=[code]` in the URL,
```

Comment:
The purpose of the coupon links is not only to add the coupon to the cart but to also add products to the cart. The coupon is just one extra thing that need to be part of the cart that will be created. We already have routes that add products to the shopping cart:

1. We have renewal links that would add an expiring plan, domain, or other product to the cart and display the checkout
2. We have plan links that would add a specific plan to the cart

If these routes check for the ?coupon=code, it can be added with the POST to the shopping cart without any server changes.

Reply:
> The purpose of the coupon links is not only to add the coupon to the cart but to also add products to the cart.

Great point, thanks! I totally missed that feature in my work so far.

> If these routes check for the ?coupon=code, it can be added with the POST to the shopping cart without any server changes.

Could you elaborate on that just a bit for me please? I'm not sure I understand what you mean exactly. POST requests to add products along with the coupon though, that sounds great to me.

Reply:
@jaswrks most of the code works with the cart by sending POST requests to `https://public-api.wordpress.com/rest/v1.1/me/shopping-cart/:site_id`, for example when you choose a plan, or when you apply a coupon on a checkout page. We can reuse the same endpoint here, there are ready-made functions in place to work with it.

Also - I can see you added some redux logic here. We want to reduxify cart, but I'd see it as a separate change. Let's just deal with flux for the purposes of this diff, transition to redux seems out of scope.